src/main/java/com/microsoft/graph/models/extensions/Multipart.java

```diff
@@ -33,56 +37,160 @@ public Multipart() {
 	 * Get the multipart boundary for use in the request header
 	 * @return the multipart boundary
 	 */
-	public String boundary() {
+	public String getBoundary() {
 		return boundary;
 	}
 	
+	/**
+	 * Set the multipart boundary for use in the request header
+	 * @param boundary The multipart boundary
+	 */
+	public void setBoundary(String boundary) {
+		this.boundary = boundary;
+	}
+	
+	/**
+	 * Get the contentType for use in the request header
+	 * @return the multipart Content-Type
+	 */
+	public String getContentType() {
+		return contentType;
+	}
+	
+	/**
+	 * Set the contentType for use in the request header
+	 * @param contentType The multipart Content-Type
+	 */
+	public void setContentType(String contentType) {
+		this.contentType = contentType;
+	}
+	
 	/**
 	 * Get the Content-Type header to send the multipart request
 	 * @return the multipart header option
 	 */
 	public HeaderOption header() {
-		return new HeaderOption("Content-Type", "multipart/form-data; boundary=\"" + boundary + "\"");
+		return new HeaderOption("Content-Type", contentType + "; boundary=\"" + boundary + "\"");
+	}
+	
+	private void writePartData(String partContent, byte[] byteArray) throws IOException{
+		out.write(partContent.getBytes(MULTIPART_ENCODING));
+		out.write(byteArray);
+		String returnContent = RETURN + RETURN;
+		out.write(returnContent.getBytes(MULTIPART_ENCODING));
 	}
 	
 	/**
-	 * Add a string part to the multipart body
+	 * Create content headers value and parameter
+	 * @param name The content header name
+	 * @param contentType The content header Content-Type
+	 * @param filename The content header filename 
+	 * @return content header value and parameter string 
+	 */
+	@VisibleForTesting String createPartHeader(String name, String contentType, String filename) {
+		StringBuilder partContent = new StringBuilder(addBoundary());
+		partContent.append("Content-Disposition: form-data");
+        if(filename != null) {
+        	if(name != null) 
+        		partContent.append("; name=\"").append(name).append("\"; filename=\"").append(filename).append("\"");
+        	else 
+        		partContent.append("; filename=\"").append(filename).append("\"");
+        }
+        else if(name != null) 
+        	partContent.append("; name=\"").append(name).append("\"");
+        if(contentType != null)
+        	partContent.append(RETURN).append("Content-Type: ").append(contentType);
+        partContent.append(RETURN).append(RETURN);
+        return partContent.toString();
+	}
+	
+	/**
+	 * Create content headers value and parameter
+	 * @param contentValue The content header value
+	 * @param contentDispParameter Map containing content paramter's key and value pair 
+	 * @return content header value and parameter string 
+	 */
+	public static String createContentHeaderValue(String contentValue, Map<String, String> contentDispParameter) {
```

Comment:
I would just drop this method.  There isn't any commonality between Content-Type and Content-Disposition values.  We can build helper methods for creating header values that are semi-colon delimited parameter lists at a later point that can be used in other places.  This method just creates confusion and we are not using it except for in tests.

Reply:
The content header's value and parameter string building is common in Content-Type and Content-Disposition. eg. 
Content-Disposition:form-data; name="someName"
Content-Type: text/html; charset="someVal"
For this common structure of building I used one helper function for both headers.